### PR TITLE
Neck Slots

### PR DESCRIPTION
**Pull request by chowdermcarthor**
:cl: Chowder McArthor
add: Added a neck slot.
tweak: Modified some of the attachment items (ties, scarves, etc) to be neck slot items instead.
/:cl:

### CODE REVIEW

**Comment by CyberbossJHCB** [comment]
This can ONLY end well

File: code/modules/clothing/neck/ties.dm [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	icon_state = "stethoscope"
+	item_color = "stethoscope"
+
+/obj/item/clothing/neck/stethoscope/attack(mob/living/carbon/human/M, mob/living/user)
+	if(ishuman(M) && isliving(user))
+		if(user.a_intent == "help")
+			var/body_part = parse_zone(user.zone_selected)
+			if(body_part)
+				var/their = "their"
+				switch(M.gender)
```
Comment (RemieRichards) [inline code/modules/clothing/neck/ties.dm]: we have pronoun procs that automatically return things such as "his","her"

Comment (chowdermcarthor) [inline code/modules/clothing/neck/ties.dm]: I didn't do this, I was copy/pasting shit from the other slots just to be certain on things I was unsure about.

Comment (RemieRichards) [inline code/modules/clothing/neck/ties.dm]: Doesn't matter, you touch code it's your job to update it :P

File: code/modules/clothing/under/ties.dm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/obj/item/clothing/tie
+/obj/item/clothing/tie //Ties moved to neck slot items, but leaving this as-is to avoid conflicts
```
Comment (chowdermcarthor) [inline code/modules/clothing/under/ties.dm]: I didn't change armbands, or medals, or the waistcoat.

**Comment by RemieRichards** [comment]
Looks ok overall, Travis is failing due to the tie path changes on metastation, so you'll have to swap those for the new paths (mostly just adding ```/neck``` from the looks of things)

File: code/modules/clothing/under/ties.dm (hunk 2 above)
Comment (chowdermcarthor) [inline code/modules/clothing/under/ties.dm]: Got it.

**Comment by chowdermcarthor** [comment]
I did it for most maps, forgot to do metastation for some reason is all.

**Comment by Nienhaus** [comment]
Chowder is cool.

**Comment by Nienhaus** [comment]
Oh and this looks cool.

**Comment by GunHog** [comment]
Pet collar :heart:

**Comment by RemieRichards** [comment]
@chowdermcarthor One thing to request, mind showing a UI shot?

**Comment by chowdermcarthor** [comment]
It's just some minor Z levels like on away missions and such now, and sure, 
https://i.gyazo.com/a14ec53b45dc89ee75baedea31dd828a.png
https://i.gyazo.com/2494a7e465393fa51badb8010df4473d.png

**Comment by RemieRichards** [comment]
Fits in nicely

**Comment by Cheridan** [comment]
So uh
what does this do that the previous system doesn't

**Comment by RemieRichards** [comment]
The previous system was shit Cheri.

**Comment by PKPenguin321** [comment]
Certain items like ties should be fine to wear no matter what, pins and medals should require a jumpsuit, idk if you've already accounted for this

**Comment by chowdermcarthor** [comment]
It's not a change to the attachment system, really. It adds a new slot, a neck slot. The reason the attachment system was mentioned was because I moved some items from the attachment system to the neck slot.

**Comment by TechnoAlchemisto** [comment]
This is fucking sick yo

**Comment by chowdermcarthor** [comment]
https://gyazo.com/876e6059af2af40e5664a3ffae776b6e
I need to remember tomorrow to go and change some of the descriptions, just noticed this while observing on bagil.

**Comment by optimumtact** [comment]
the tie inventory slot icon looks like a penis

File: code/modules/clothing/under/ties.dm [hunk 3]
```diff
@@ -55,7 +55,7 @@
 
 /obj/item/clothing/tie/proc/on_uniform_dropped(obj/item/clothing/under/U)
 	return
-
+/*
```
Comment (optimumtact) [inline code/modules/clothing/under/ties.dm]: remove commented out code

**Comment by Cheridan** [comment]
@RemieRichards not an argument

**Comment by optimumtact** [comment]
It's a lot less confusing for new players

**Comment by Shapsy** [comment]
Attachments were snowflakey and annoying to deal with to the point that no one ever really bothered using any, now they would work like any other item slot

**Comment by XDTM** [comment]
Can't wait for exploding collars

**Comment by Cheridan** [comment]
@Shapsy Accessory items still exist though, so any flaws with THAT system still exist on top of having a new slot taking screen real estate. this isn't a replacement, it's an addition.

**Comment by AnturK** [comment]
I believe abstract "accessory" slot would be better for all the "does not really fit any category" equipment like ties,holsters and similar. Not sure how handling of layers would work in that case though.

File: code/__DEFINES/clothing.dm [hunk 4]
```diff
@@ -98,7 +102,8 @@
 #define HAND_LEFT	512
 #define HAND_RIGHT	1024
 #define HANDS		1536
-#define FULL_BODY	2047
+#define NECK		2048
+#define FULL_BODY	4096 //2047
```
Comment (X-TheDark) [inline code/__DEFINES/clothing.dm]: This needs to be 4095 to fulfill the functionality of the old flag. Here's the reason:
* Old FULL_BODY corresponded to bitflag (111 1111 1111 - meaning it's considered to cover every bodypart);
* New FULL_BODY corresponds to bitflag (1 0000 0000 0000), which mean it **no longer is considered to cover the entire body (or anything, really), logic-wise**;

Comment (chowdermcarthor) [inline code/__DEFINES/clothing.dm]: Ah, so that's why it was that number, I wasn't sure why it was that, I'll change it as soon as I can.

**Comment by PKPenguin321** [comment]
But yeah big :+1: to this, makes coding special neck items much easier

**Comment by Tacolizard** [comment]
finally i can wear a medal and scarf without being weighed down

**Comment by RemieRichards** [comment]
@Cheridan moving the majority of the items under that system to this (which this PR does, neck items were atleast 70% of accessories) increases the odds we'll be able to get RID OF the shit system.
cmon now cheri you're better than this.

**Comment by Iamgoofball** [comment]
What if accessories was a mini inventory with 3 slots

**Comment by Tacolizard** [comment]
@Iamgoofball that sounds cool.

also should medals be pinned on like right now or be neck slots? because right now they're more like pins.

**Comment by chowdermcarthor** [comment]
Gah fuck, didn't mean to upload that to this.

**Comment by GunHog** [comment]
Write a commit to revert your most recent change.

**Comment by PKPenguin321** [comment]
git checkout origin/master -- [filename]

gets the file from the main repo and replaces your local file with it. use this on the unwanted changed file and commit

**Comment by Anonmare** [comment]
:+1:
Also, question, will this allow for things like cloaks? Since they do slot around the neck and all.

**Pull request merged**